src/system-health/scripts/healthd

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python2
+
+"""
+    healthd
+    System health monitor daemon for SONiC
+"""
+
+import signal
+import threading
+
+from sonic_py_common.daemon_base import DaemonBase
+from sonic_py_common.logger import Logger
+from swsssdk import SonicV2Connector
+
+from health_checker.manager import HealthCheckerManager
+
+SYSLOG_IDENTIFIER = 'healthd'
+logger = Logger(SYSLOG_IDENTIFIER)
+
+
+class HealthDaemon(DaemonBase):
+    """
+    A daemon that run as a service to perform system health checker with a configurable interval. Also set system LED 
+    according to the check result and store the check result to redis.
+    """
+    SYSTEM_HEALTH_TABLE_NAME = 'SYSTEM_HEALTH_INFO'
+
+    def __init__(self):
+        """
+        Constructor of HealthDaemon.
+        """
+        DaemonBase.__init__(self, SYSLOG_IDENTIFIER)
+        self._db = SonicV2Connector(host="127.0.0.1")
+        self._db.connect(self._db.STATE_DB)
+        self.stop_event = threading.Event()
+
+    def deinit(self):
+        """
+        Destructor. Remove all entries in $SYSTEM_HEALTH_TABLE_NAME table.
+        :return: 
+        """
+        self._clear_system_health_table()
+
+    def _clear_system_health_table(self):
+        self._db.delete_all_by_pattern(self._db.STATE_DB, HealthDaemon.SYSTEM_HEALTH_TABLE_NAME)
+
+    # Signal handler
+    def signal_handler(self, sig, frame):
+        """
+        Signal handler
+        :param sig: Signal number
+        :param frame: not used
+        :return:
+        """
+        if sig == signal.SIGHUP:
+            logger.log_info("Caught SIGHUP - ignoring...")
+        elif sig == signal.SIGINT:
+            logger.log_info("Caught SIGINT - exiting...")
+            self.stop_event.set()
+        elif sig == signal.SIGTERM:
+            logger.log_info("Caught SIGTERM - exiting...")
+            self.stop_event.set()
+        else:
+            logger.log_warning("Caught unhandled signal '" + sig + "'")
```

Comment:
Signal handler should be inherited from DaemonBase. Unless you want to override it, this should not be necessary.

Reply:
We override it by set a stop event here. And after the event was set, the infinite loop will exit and do some clear operation.